notebooks/working/6.0-mab-estimate-causal-effect.md

```diff
@@ -0,0 +1,2197 @@
+                                        new_alt_id_name=custom_alt_id)
+    
+    # Create a cars per licensed drivers column
+    long_sim_data["cars_per_licensed_drivers"] = 0
+    long_sim_data.loc[long_sim_data.num_licensed_drivers > 0,
+                      "cars_per_licensed_drivers"] = long_sim_data.num_cars / long_sim_data.num_licensed_drivers.astype(float)
+
+    # Add a variable representing cost divided by distance
+    long_sim_data["cost_per_distance"] = 0
+    long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                      "cost_per_distance"] = (long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                                                                "total_travel_cost"] /
+                                              long_sim_data.loc[long_sim_data.mode_id.isin([1, 2, 3]),
+                                                        "total_travel_distance"])
+    print('Simulating Choices...')
+    # Calculate probabilities for each alternative
+    # based on the estimated model
+    posterior_probs = mnl_model.predict(long_sim_data)
+    
+    
+    # Simulate choice data
+    long_sim_data['sim_choice'] = viz.simulate_choice_vector(posterior_probs,
+                               long_sim_data['observation_id'].values)
+    
+    
+    print('Estimating the choice model...')
+    # Estimate the basic MNL model, using the hessian and newton-conjugate gradient
+    mnl_model_sim = pl.create_choice_model(data=long_sim_data,
+                                           alt_id_col=alternative_id_col,
+                                           obs_id_col=observation_id_col,
+                                           choice_col=choice_column,
+                                           specification=mnl_specification,
+                                           model_type="MNL",
+                                           names=mnl_names)
+
+    num_vars = len(reduce(lambda x, y: x + y, mnl_names.values()))
+    # Note newton-cg used to ensure convergence to a point where gradient 
+    # is essentially zero for all dimensions. 
+    mnl_model_sim.fit_mle(np.zeros(num_vars),
+                          method="BFGS")
+    
+    mnl_model_sim.coefs.name = 'simulation_' + str(number)
+
+    models_dictionary[number] = mnl_model_sim
+    
+    print('Simulation number', number , 'is complete!')
+    print('==========================================')
+    print('==========================================')
+```
+
+# 5. Plotting results of the simulation 
+
+```python
+# Creating a dataframe for model parameters
+sim_estimated_par = pd.concat([models_dictionary[number].coefs for number in models_dictionary.keys()], axis=1).T
+sim_estimated_par.reset_index(drop=True,inplace=True)
+sim_estimated_par.head()
+```
+
+```python
+# Loop around the different parameters
+# and plot the figures:
+
+for par in sim_estimated_par.columns:
+    PlotParams(sim_estimated_par[par],mnl_model, (10,6))
```

Comment:
We definitely don't want to plot the estimated parameters from each simulation.

We are interested in the causal effects not the estimated parameters per say.

If you create the arrays of naive, estimated, and true causal effects as described in my comment above, those should be plotted here. It should be a single call to seaborns distplot or kdeplot, no need for a loop when plotting.